packages/test/test-end-to-end-tests/src/mocking.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { isPromiseLike } from "@fluidframework/core-utils";
+
+export type UnPromise<T> = T extends Promise<infer U> ? U : T;
```

Comment:
What about just using `Awaited`?  https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
nice. i think this code was written before it that existed, as this is just moved from another file to be reused

Reply:
I thought it was a more refined / flexible version of what you have here. e.g. deals with multiple / mismatched `Promise`.  Does it not work out to use it here?

Reply:
i'm getting type error when i try to use it, and looking a the definition i think the nesting nature it the culprit, i only want to strip off the top level promise if it exists